src/content/docs/node/introduction.mdx

```diff
@@ -1,3 +1,502 @@
+            - Install it from the [official website](https://code.visualstudio.com/).
+
+        **Recommended extensions**:
+            - **Node.js Extension Pack**: Adds debugging, IntelliSense, and tools for Node.js.
+            - **ESLint**: Helps in identifying and fixing coding errors.
+            - **Prettier**: Automatically formats your code.
+
+## 3. Version Control: Git
+
+    Version control helps you manage code changes effectively.
+        - Install Git from the official website.
+        - Set up Git in your project directory:
+         ```bash
+        git init
+        git config --global user.name "Your Name"
+        git config --global user.email "your.email@example.com"
+        ```
+
+## 4. Project Setup
+
+    Start by creating a new Node.js project:
+
+    <Steps>
+
+    1. **Download Node.js**:
+
+        ```bash
+        mkdir my-node-project
+        cd my-node-project
+        ```
+    2. **Initialize the project**:
+
+        ```bash
+        npm init -y
+        ```
+
+    </Steps>
+
+## 5. Useful NPM Packages
+
+    Some essential packages to install right away:
+        - Express: A minimalist web framework to install use **npm install express**.
+        - Nodemon: Automatically restarts the server when you make changes use **npm install --save-dev nodemon**.
+        - dotenv: Manages environment variables **npm install dotenv**.
+
+## 6. Folder Structure
+
+    A good folder structure keeps your code organized.
+
+        <FileTree>
+
+        - /my-node-project
+            - /config
+              - /db.js
+            - /controllers
+              - /userController.js
+            - /models
+              - /userModel.js
+            - /routes
+              - /route.js
+          - .env
+          - index.js
+          - package.json
+
+        </FileTree>
+
+## Creating Your First Node.js Application: A Step-by-Step Tutorial
+
+---
+
+### Steps to Create Your First Node.js Project
+
+1. **Create a folder** on your system and open it in your code editor.
+2. **Open the terminal** and run `npm init -y`; this will add a new file called `package.json`.
+3. **Create a file named `index.js`** and add your JavaScript code.
+4. **To see the output**, add `"start": "node index.js"` in the `"scripts"` section of `package.json`, then run `npm start` in the terminal.
+
+```js
+// index.js
+console.log("Hello, Node.js!");
+```
+
+Inside your project folder, create a file called index.js, which will serve as your main application file. To set up a basic Express server, start by requiring the Express module and configuring a simple server. In the index.js file, import Express and define a route for the homepage that responds with a message. Finally, set the server to listen on a specified port, such as 3000, and log a message indicating that the server is running. This establishes the foundation for your Node.js application using Express.
+
+### First Node.js Project with express
```

Comment:
here add one more code snippet for installing express module before the express code snippet